File: rts/Rendering/Env/Particles/Generators/EmgParticleGenerator.cpp
#include "EmgParticleGenerator.h"

CR_BIND(EmgParticleData, )
CR_REG_METADATA(EmgParticleData,
(
	CR_MEMBER(pos),
	CR_MEMBER(drawRadius),
	CR_MEMBER(speed),
	CR_MEMBER(createFrame),
	CR_MEMBER(animParams),
	CR_MEMBER(color),
	CR_MEMBER(rotParams),
	CR_MEMBER(drawOrder),
	CR_MEMBER(texCoord)
))

File: cont/base/springcontent/shaders/GLSL/Particles/EmgParticleGenerator.lua
return {
	MainCode =
[[
	float currTime = frameInfo.x + frameInfo.y - createFrame;

	SetCurrentAnimation(animParams, currTime);
	vec3 drawPos = pos + speed * frameInfo.y;
	
	// rotParams do we need it?

	AddEffectsQuadCamera(
		drawOrder,
		animParams,
		drawPos, vec2(drawRadius), texCoord,
		color
	);
]]
}